cpp/oneapi/dal/backend/primitives/sparse_blas/test/gemm_dpc.cpp

```diff
@@ -0,0 +1,256 @@
+                column_indices_host[j] = column_index;
+            }
+        }
+
+        auto row_offsets_host_ary = dal::array<std::int64_t>::empty(local_m + 1);
+        std::int64_t * row_offsets_host = row_offsets_host_ary.get_mutable_data();
+
+        /// Initialize row offsets of the even rows
+        for(std::int64_t i = 0; i < local_m + 1; i += 2) {
+            row_offsets_host[i] = (i / 2) * local_k + indexing_offset;
+        }
+        /// Initialize row offsets of the odd rows
+        for(std::int64_t i = 1; i < local_m + 1; i += 2) {
+            row_offsets_host[i] = value_count_in_odd_rows + (i / 2) * local_k + indexing_offset;
+        }
+
+        std::int64_t * column_indices = column_indices_ary_.get_mutable_data();
+        auto column_indices_event = q.submit([&](sycl::handler& cgh) {
+            cgh.memcpy(column_indices, column_indices_host, element_count * sizeof(std::int64_t));
+        });
+
+        std::int64_t * row_offsets = row_offsets_ary_.get_mutable_data();
+        auto row_offsets_event = q.submit([&](sycl::handler& cgh) {
+            cgh.memcpy(row_offsets, row_offsets_host, (local_m + 1) * sizeof(std::int64_t));
+        });
+
+        return set_csr_data(q,
+                            a,
+                            local_m,
+                            local_k,
+                            indexing,
+                            data,
+                            column_indices,
+                            row_offsets,
+                            { data_event, column_indices_event, row_offsets_event });
+    }
+
+    auto B() {
+        check_if_initialized();
+        if (bo == ndorder::c) {
+            return ndarray<float_t, 2, bo>::ones(this->get_queue(), { k_, p_ }, alloc_);
+        }
+        return ndarray<float_t, 2, bo>::ones(this->get_queue(), { p_, k_ }, alloc_);
+    }
+
+    auto C() {
+        check_if_initialized();
+        if (co == ndorder::c) {
+            return ndarray<float_t, 2, co>::empty(this->get_queue(), { m_, p_ }, alloc_);
+        }
+        return ndarray<float_t, 2, co>::empty(this->get_queue(), { p_, m_ }, alloc_);
+    }
+
+    void test_gemm() {
+        sparse_matrix_handle a;
+        auto a_e = A(a);
+        auto [b, b_e] = B();
+        auto c = C();
+
+        gemm(this->get_queue(), trans_a, a, b, c, { a_e, b_e }).wait_and_throw();
+
+        check_matmul(c);
+    }
+
+    void check_matmul(const ndarray<float_t, 2, co>& mat) {
+        check_if_initialized();
+        REQUIRE(mat.get_shape() == ndshape<2>{ m_, p_ });
+
+        auto mat_host = mat.to_host(this->get_queue());
+        const float_t* mat_ptr = mat_host.get_data();
+        const std::int64_t result_even = (k_ + 1) / 2;    // even rows result
+        const std::int64_t result_odd = k_ / 2;           // odd rows result
+
+        for (std::int64_t i = 0; i < m_; i += 2) {
+            for (std::int64_t j = 0; j < p_; j++) {
```

Comment:
It is highly discouraged to use postfix operators. In this particular case it won't make you any trouble, but in some instances (with custom and complex iterators) it can result in generating inefficient code